### PR TITLE
RFC: Throw an error when coalesce is passed only nothing/missing arguments

### PR DESCRIPTION
**Pull request by nalimilan**
Remove the one-argument method, which does not allow passing a replacement value which can never be nothing/missing. The old behavior can still be obtained by passing `Some(nothing)`/`Some(missing)` as last argument.

Possible fix for https://github.com/JuliaLang/julia/issues/26927.

The advantage of this PR is that it implements a strict subset of the previous behavior. So if we change our minds and decide to allow returning `nothing`/`missing` when all arguments are `nothing`/`missing` (as in SQL and dplyr), it won't break anything.

EDIT: instead of throwing an error, we could first print a deprecation.

EDIT2: the last commit also implements the proposal made by @Keno at https://github.com/JuliaLang/julia/issues/26927, i.e. when `nothing` and `missing` are mixed only the first one is replaced, and the other is considered as non missing and therefore returned as-is.

### CODE REVIEW

File: base/some.jl [hunk 1]
```diff
@@ -62,13 +76,20 @@ julia> coalesce(nothing, Some(1))
...
-coalesce(x::Some, y...) = x.value
-coalesce(x::Union{Nothing, Missing}, y...) = coalesce(y...)
+coalesce(x::Nothing, y) = y
+coalesce(x::Missing, y) = y
+coalesce(x::Nothing, y::Some) = y.value
+coalesce(x::Missing, y::Some) = y.value
+for T in (:Nothing, :Missing)
+    @eval begin
+        coalesce(x::$T, y::Union{Nothing, Missing}) =
+            throw(ArgumentError("coalesce requires that least one argument differs from `nothing` or `missing`; " *
```
Comment (ararslan) [inline base/some.jl]: The line here is pretty long. I'd suggest changing the semicolon to a period and the space after it to a newline. Not a big deal though.

Comment (nalimilan) [reply base/some.jl]: Fast! I wasn't sure what's the convention about this. It's indeed problematic in the doctest.

Comment (nalimilan) [reply base/some.jl]: Fixed.

**Comment by bkamins** [comment]
If this PR is merged I would recommend to update documentation of `Some` in this PR and clearly state that it is used also as a wrapper type for `missing` (not only `nothing`).

**Comment by JeffBezanson** [comment]
I think this is going in the right direction, but it seems really weird to me that `coalesce(missing, missing)` is an error, but `coalesce(nothing, missing)` is totally fine. The latter call is much harder to make sense of.

**Pull request closed without merging**